tests/std/tests/P0408R7_efficient_access_to_stringbuf_buffer/test.cpp

```diff
@@ -0,0 +1,177 @@
+        allocator.deallocate(p, n);
+    }
+
+    counting_allocator() {}
+
+    template <typename U>
+    explicit counting_allocator(const counting_allocator<U>& c) : count(c.count) {}
+};
+
+template <typename T, typename U>
+bool operator==(const counting_allocator<T>& c1, const counting_allocator<U>& c2) {
+    return c1.allocator == c2.allocator;
+}
+
+using counting_string        = std::basic_string<char, std::char_traits<char>, counting_allocator<char>>;
+using counting_stringstream  = std::basic_stringstream<char, std::char_traits<char>, counting_allocator<char>>;
+using counting_istringstream = std::basic_istringstream<char, std::char_traits<char>, counting_allocator<char>>;
+using counting_ostringstream = std::basic_ostringstream<char, std::char_traits<char>, counting_allocator<char>>;
+
+template <typename Stream>
+struct test_counting_allocator {
+    void operator()(counting_string s) {
+        Stream stream{move(s)};
+        s = move(stream).str();
+        stream.str(move(s));
+        if constexpr (!std::is_same_v<Stream, counting_istringstream>) {
+            assert(stream.rdbuf()->get_allocator().count == 1);
+        }
+    }
+};
+
+template <template <typename> typename Test>
+void run_counting_test() {
+    run_test_util<Test<counting_stringstream>>();
+    run_test_util<Test<counting_istringstream>>();
+    run_test_util<Test<counting_ostringstream>>();
+}
+
+template <const char* str>
+void test_iterator_dereference_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    (void) *iterator; // cannot dereference invalid iterator
+}
+
+template <const char* str>
+void test_iterator_operator_arrow_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    (void) iterator.operator->(); // cannot dereference invalid iterator
+}
+
+template <const char* str>
+void test_iterator_increment_death() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    ++iterator; // cannot increase invalid iterator
+}
+
+template <const char* str>
+void test_iterator_increment_zero() {
+    string s      = str;
+    auto iterator = s.begin();
+    stringstream stream{move(s)};
+    iterator += 0; // OK
+}
+
+int main(int argc, char* argv[]) {
+    std_testing::death_test_executive exec([] {
+        run_test<test_rvalue>();
+        run_test<test_allocator>();
+        run_counting_test<test_counting_allocator>();
+
+        test_iterator_increment_zero<small_string>();
+        test_iterator_increment_zero<large_string>();
+    });
+
+#if _ITERATOR_DEBUG_LEVEL != 0
+    exec.add_death_tests({test_iterator_dereference_death<small_string>, test_iterator_dereference_death<large_string>,
+        test_iterator_operator_arrow_death<small_string>, test_iterator_operator_arrow_death<large_string>,
+        test_iterator_increment_death<small_string>, test_iterator_increment_death<large_string>});

```

Comment:
Now that we've fixed the CI system, this test appears to be failing with:
```
running normal tests... passed!
running death test 0... returned 0xC0000409, a failure code (this is good)
running death test 1... returned 0xC0000409, a failure code (this is good)
running death test 2... returned 0xC0000409, a failure code (this is good)
running death test 3... returned 0xC0000409, a failure code (this is good)
running death test 4... returned 0xC0000409, a failure code (this is good)
running death test 5... returned 0, a success code (this is bad)
Terminate!
```

Reply:
Interesting, I thought the result of x86 and x64 builds should be the same. I'll have a look.